lib/components/base-components/NormalComponent.ts

```diff
@@ -373,6 +375,7 @@ export class NormalComponent<
   }
 
   _doInitialSchematicComponentRenderWithSymbol() {
+    if (this.root?.schematicDisabled) return
```

Comment:
I think rather checking the `schematicDisabled` in each and every function. You can check it in the `runRenderPhase` function,  where all the function having the prefix as `Schematic` can be skipped.

cc: @seveibar

Reply:
@imrishabh18 so this makes sense but i actually like the approach where we do it for each function. The reason why is we’re eventually transitioning to a “dependency graph”-style render pipeline, so id prefer to keep Renderable simple until we make that transition